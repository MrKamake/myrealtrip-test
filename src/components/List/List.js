import React, { Component } from 'react';
import './List.scss';
import { PRODUCT_THUMNAIL_PATH } from '../../constants/constant';
import { IoMdHeartEmpty } from 'react-icons/io';

class List extends Component {
  componentDidMount() {
    this.props.onLoad();
  }

  render() {
    const { productList } = this.props;

    return (
      <div className="List">
        {productList.map((product, index) => (
          <div className="product-element" key={index}>
            <img
              className="product-thumnail"
              src={`${PRODUCT_THUMNAIL_PATH}${index + 1}.jpg`}
            />
            <div className="wish-icons">
              <IoMdHeartEmpty />
            </div>
            <div className="product-title">{product.title}</div>
            <div className="product-price">{product.price}</div>
          </div>
        ))}
      </div>
    );
  }
}

export default List;
