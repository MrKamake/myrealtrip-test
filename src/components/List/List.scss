@import '../../styles/utils.scss';

@media only screen and (min-width: 1280px) and (max-width: 1347px) {
  .List {
    padding: 30px 5%;
  }
}
@media only screen and (min-width: 1348px) and (max-width: 1680px) {
  .List {
    padding: 30px 12%;
  }
}

.List {
  position: relative;
  display: inline-flex;
  flex-wrap: wrap;
  justify-content: space-around;
  padding-top: 100px;
  .empty-wish-list {
    position: relative;
    padding: 300px 470px;
    font-size: 1.4rem;
    color: darkgray;
  }
  .product-card {
    position: relative;
    margin: 30px 2px;
    width: 250px;
    height: 280px;
    border: 1px solid $lightgray;
    border-radius: 4px;
    .product-module-thumnail {
      height: 166px;
      overflow: hidden;
      .product-thumnail {
        width: 100%;
        height: 100%;
      }
    }
    .product-module-body {
      padding: 16px;
      .product-title {
        display: -webkit-box;
        margin: 0;
        width: 100%;
        line-height: 1.47;
        font-size: 15px;
        font-weight: 600;
        color: $darkblueblack;
        overflow: hidden;
        -webkit-box-orient: vertical;
        -webkit-line-clamp: 2;
      }
      .product-price {
        position: absolute;
        bottom: 8%;
        color: $lightbluegray;
        font-size: 14px;
        font-weight: 500;
      }
    }
    .product-wish-icons {
      cursor: pointer;
      position: absolute;
      top: 8px;
      right: 8px;
      .heart-icon {
        color: $lovelypink;
      }
      .heart-empty-icon {
        color: white;
      }
    }
  }
}
