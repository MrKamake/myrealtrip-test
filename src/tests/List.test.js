import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import List from '../components/List/List';

configure({ adapter: new Adapter() });

describe('<List />', () => {
  let wrapper, props;

  beforeEach(() => {
    props = {
      onLoad: jest.fn(),
      productList: [
        {
          id: 4,
          title: '산에서 바라본 시계탑',
          price: 22000
        },
        {
          id: 5,
          title: '길거리 그림들',
          price: 50000
        },
        {
          id: 6,
          title: '표지판',
          price: 7000
        }
      ],
      selectedListMessage: 'default',
      updateWishList: jest.fn(),
      wishList: []
    };

    wrapper = shallow(<List {...props} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain elements', () => {
    const listComponent = wrapper.find('.List');
    expect(listComponent.length).toBe(1);

    const productCard = wrapper.find('.product-card');
    expect(productCard.length).toBe(3);

    expect(wrapper.contains(<h3 className="product-title">표지판</h3>)).toBe(
      true
    );
    expect(
      wrapper.contains(<h3 className="product-title">길거리 그림들</h3>)
    ).toBe(true);
    expect(wrapper.contains(<div className="product-price">7,000원</div>)).toBe(
      true
    );
  });

  it('should contain Empty class name', () => {
    props.selectedListMessage = 'wish';
    wrapper = shallow(<List {...props} />);

    const emptyWishList = wrapper.find('.Empty');
    expect(emptyWishList.length).toBe(1);
  });
});
