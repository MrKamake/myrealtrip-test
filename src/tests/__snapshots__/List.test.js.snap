// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> matches snapshot 1`] = `
<div
  className="List "
>
  <div
    className="product-card "
    key="0"
  >
    <div
      className="product-module-thumnail"
    >
      <img
        alt="product thumnail"
        className="product-thumnail"
        src="https://myrealtrip-upload-us.s3.amazonaws.com/share/thumbnail/thumbnail-4.jpg"
      />
    </div>
    <div
      className="product-module-body"
    >
      <h3
        className="product-title"
      >
        산에서 바라본 시계탑
      </h3>
      <div
        className="product-price"
      >
        22,000
        원
      </div>
    </div>
    <div
      className="product-wish-icons"
      onClick={[Function]}
    >
      <FaRegHeart
        className="heart-empty-icon"
      />
    </div>
  </div>
  <div
    className="product-card "
    key="1"
  >
    <div
      className="product-module-thumnail"
    >
      <img
        alt="product thumnail"
        className="product-thumnail"
        src="https://myrealtrip-upload-us.s3.amazonaws.com/share/thumbnail/thumbnail-5.jpg"
      />
    </div>
    <div
      className="product-module-body"
    >
      <h3
        className="product-title"
      >
        길거리 그림들
      </h3>
      <div
        className="product-price"
      >
        50,000
        원
      </div>
    </div>
    <div
      className="product-wish-icons"
      onClick={[Function]}
    >
      <FaRegHeart
        className="heart-empty-icon"
      />
    </div>
  </div>
  <div
    className="product-card "
    key="2"
  >
    <div
      className="product-module-thumnail"
    >
      <img
        alt="product thumnail"
        className="product-thumnail"
        src="https://myrealtrip-upload-us.s3.amazonaws.com/share/thumbnail/thumbnail-6.jpg"
      />
    </div>
    <div
      className="product-module-body"
    >
      <h3
        className="product-title"
      >
        표지판
      </h3>
      <div
        className="product-price"
      >
        7,000
        원
      </div>
    </div>
    <div
      className="product-wish-icons"
      onClick={[Function]}
    >
      <FaRegHeart
        className="heart-empty-icon"
      />
    </div>
  </div>
</div>
`;
